src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapFloatVectorTransfer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.apache.commons.lang.StringUtils;
+import org.opensearch.knn.index.vectorvalues.KNNFloatVectorValues;
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Transfer float vectors to off heap memory.
+ */
+public final class OffHeapFloatVectorTransfer extends OffHeapQuantizedVectorTransfer<float[], float[]> {
```

Comment:
If we have to convert and transfer a float[] to byte[] do we create a new class for that kind of transfer? This is the reason I think 1 generic is good, but other generic I am not sure how useful it is.

Reply:
For now float[] to float[] does not have a use case so kept it as it is

Reply:
Sure will check.

But in my thought process on this `OffHeapFloatVectorTransfer` should just say I am going to transfer a batch of float vectors to a memory address. How those floats are getting created it should not know about it.

Please let me know what you think here. Since you are more close to the code than I am.